src/main/java/mate/academy/springboot/swagger/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.springboot.swagger.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.repository.ProductRepository;
+import mate.academy.springboot.swagger.service.ProductService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    @Override
+    public Product add(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productRepository.getById(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public Product update(Product product) {
+        Product productFromDb = get(product.getId());
+        productFromDb.setTitle(product.getTitle());
+        productFromDb.setPrice(product.getPrice());
+        return add(productFromDb);
```

Comment:
use the repository method `save` directly, cause if something changes in the method `add`, it could be affected your `update` method